### DISCUSSION TITLE
new() or new ()

### DISCUSSION THREAD

**Discussion by twojnarowski** [codecadwallader/codemaid#937] [Polls]
Codemaid cleans my code to leave no space after a new keyword, but [StyleCop suggests](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/SA1000.md) that there should be a space after the new keyword.
Which way is actually the preferred way? Or is this completely up to somebody's personal taste?

**Comment by codecadwallader** [comment]
CodeMaid invokes Visual Studio's native formatting which I suspect is where this space is being removed.  If you invoke Edit->Advanced->Format Document do you see the same behavior?

Looking at the StyleCop documentation linked above it seems like it depends on the context of where it's being used:

> The new and stackalloc keywords should always be followed by a space, except in the following cases:
> The new or stackalloc keyword is used to create a new implicitly-typed array. In this case there should be no space between the keyword and the opening array bracket.
> The new keyword is part of a generic type constraint. In this case there should be no space between the new keyword and the opening parenthesis.

**Comment by twojnarowski** [comment]
If i have the following code:
```
new ();
new();
```
CodeMaid will change it to:
```
new();
new();
```
And invoking Format Document as you have mentioned changes the initial code to:
```
new();
new();
```
So it works the same as CodeMaid. Now I can see that in [Microsoft Docs](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/new-operator#constructor-invocation) the new keyword for creating new objects omitting the type name is written without the space. It leads to the conclusion, that StyleCop's rules have not been updated to cover the case for this kind of usage of new operator introduced in C# 9.0

**Reply by twojnarowski** [reply]
So I have brought this to their attention with an [issue](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/issues/3554) and my question is resolved here, in this case new() is the correct form and that is not an issue casued by CodeMaid.
